acceptance-tests/cloudsql_mysql_migration_test.go

```diff
@@ -0,0 +1,116 @@
+	"fmt"
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("MySQL service instance migration", Label("mysql-data-migration"), func() {
+	It("allows access to migrated data", func() {
+		By("creating a service broker with matching config")
+		const mysql56plan = `[{"name":"default","id":"eec62c9b-b25e-4e65-bad5-6b74d90274bf","description":"MySQL v5.6 10GB storage","mysql_version":"MYSQL_5_6","storage_gb":10,"tier":"db-n1-standard-2"}]`
+
+		serviceBroker := brokers.Create(
+			brokers.WithPrefix("csb-mysql"),
+			brokers.WithEnv(apps.EnvVar{Name: brokers.PlansMySQLVar, Value: mysql56plan}),
+		)
+		defer serviceBroker.Delete()
+
+		By("asynchronously starting the target service instance creation")
+		databaseName := random.Name(random.WithPrefix("migrate-database"))
+		targetServiceInstance := services.CreateInstance(
+			"csb-google-mysql",
+			"default",
+			services.WithParameters(map[string]any{
+				"db_name":                    databaseName,
+				"public_ip":                  false,
+				"backups_retain_number":      0,
+				"allow_insecure_connections": true, // MySQL 5.6 does not support TLS bindings
+			}),
+			services.WithBroker(serviceBroker),
+			services.WithAsync(),
+		)
+
+		By("creating the source service instance")
+		const sourceServiceOffering = "google-cloudsql-mysql-vpc"
+		const sourceServicePlan = "default"
+		sourceServiceInstance := services.CreateInstance(
+			sourceServiceOffering,
+			sourceServicePlan,
+			services.WithBrokerName(legacyBrokerName),
+			services.WithParameters(map[string]any{
+				"tier":            legacyDBTier,
+				"private_network": os.Getenv("GCP_PAS_NETWORK"),
+				"database_name":   databaseName,
+				"backups_enabled": "false",
+				"binlog":          "false",
+			}),
+		)
+		defer sourceServiceInstance.Delete()
+
+		By("binding an app to the source service instance")
+		sourceApp := apps.Push(apps.WithApp(apps.MySQL))
+		defer sourceApp.Delete()
+
+		sourceInstanceBinding := sourceServiceInstance.Bind(sourceApp)
+		sourceApp.Start()
+
+		By("adding some data in the source database")
+		key := random.Hexadecimal()
+		value := random.Hexadecimal()
+		sourceApp.PUT(value, "/key-value/%s", key)
+
+		By("waiting for the target service creation to succeed")
+		services.WaitForInstanceCreation(targetServiceInstance.Name)
+		defer targetServiceInstance.Delete()
+
+		By("extracting a service key for the source service instance")
+		credentials := sourceInstanceBinding.Credential()
+		legacyBinding := legacybindings.ExtractLegacyBinding(credentials)
+
+		By("creating a bucket")
+		bucketName := "bucket-" + databaseName
+		gsql.CreateBackupBucket(bucketName)
+		defer gsql.DeleteBucket(bucketName)
+
+		By("performing the backup")
+		backupURI := gsql.CreateBackup(legacyBinding.InstanceName, databaseName, bucketName)
+
+		By("preparing the restore in the target service instance")
+		targetInstanceIaaSName := fmt.Sprintf("csb-mysql-%s", targetServiceInstance.GUID())
```

Comment:
thoughts: it would be better to pass the `instance_name` as a parameter when creating the instance. It will avoid future problems if we change the default value for the property.

Reply:
I agree this would make the test more stable, but I prefer this way of doing it because it matches the documentation step that says:
```
Retrieve the GCP database instance ID for the new service instance.
   The GCP database instance ID is the Cloud Foundry service instance GUID with the prefix
   `csb-mysql-` added to it. This is a string similar to `csb-mysql-XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX`.
```